cudaextender/src/ungapped_xdrop_kernels.cuh

```diff
@@ -60,21 +60,50 @@ __global__ void compress_output(const int32_t* d_done,
                                 int num_hits);
 
 // Binary predicate for sorting the ScoredSegmentPairs
+struct scored_segment_pair_equal
+{
+    __host__ __device__ bool operator()(ScoredSegmentPair x, ScoredSegmentPair y)
+    {
+            return ( ( 
+                        (x.seed_pair.target_position_in_read - x.seed_pair.query_position_in_read) == (y.seed_pair.target_position_in_read - y.seed_pair.query_position_in_read) 
+                        ) 
+
+                    &&  
+
+                    ( 
+                     ( 
+                      (x.seed_pair.target_position_in_read >= y.seed_pair.target_position_in_read) 
+                      && 
+                      ( (x.seed_pair.target_position_in_read + x.length) <= (y.seed_pair.target_position_in_read + y.length) )  
+                      ) 
+
+                     || 
+
+                     ( 
+                      ( y.seed_pair.target_position_in_read >= x.seed_pair.target_position_in_read ) 
+                      && 
+                      ( (y.seed_pair.target_position_in_read + y.length) <= (x.seed_pair.target_position_in_read + x.length) ) 
+                      ) 
+                     ) 
+                    );
+    }
+};
+
 struct scored_segment_pair_comp
 {
     __host__ __device__ bool operator()(const ScoredSegmentPair& x, const ScoredSegmentPair& y)
     {
-        if (x.seed_pair.query_position_in_read < y.seed_pair.query_position_in_read)
+        if ( (x.seed_pair.target_position_in_read - x.seed_pair.query_position_in_read) < ( y.seed_pair.target_position_in_read - y.seed_pair.query_position_in_read))
             return true;
-        else if (x.seed_pair.query_position_in_read == y.seed_pair.query_position_in_read)
+        else if ( (x.seed_pair.target_position_in_read - x.seed_pair.query_position_in_read) == ( y.seed_pair.target_position_in_read - y.seed_pair.query_position_in_read))
         {
-            if (x.length > y.length)
+            if (x.seed_pair.target_position_in_read < y.seed_pair.target_position_in_read)
                 return true;
-            else if (x.length == y.length)
+            else if (x.seed_pair.target_position_in_read == y.seed_pair.target_position_in_read)
             {
-                if (x.seed_pair.target_position_in_read < y.seed_pair.target_position_in_read)
+                if (x.seed_pair.query_position_in_read > y.seed_pair.query_position_in_read)
```

Comment:
If the shift/difference `(*.seed_pair.target_position_in_read - *.seed_pair.query_position_in_read)` is equal for `x` and `y`, and the `*.seed_pair.target_position_in_read` is equal for them, then `*.seed_pair.query_position_in_read` must be equal as well. Hence this comparison can be skipped.
Maybe it was meant to compare `x.length` and `y.length`? Or is that not needed anymore?